File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        vote = 0
        candidate = -1

        for i in range(len(nums)):
            if vote == 0:
                candidate = nums[i]
                vote = 1

            else:
                if candidate == nums[i]:
                    vote += 1
                else:
                    vote -= 1

        
        count = 0

        for i in range(len(nums)):
            if candidate == nums[i]:
                count +=1


        if count > (len(nums)//2):
            return candidate
        else:
            return -1

File: 169-majority-element/majority-element.ts
function majorityElement(nums: number[]): number {
    let candidate: number = -1
    let votes: number = 0

    for(let i of nums){
        if (votes == 0){
            candidate = i;
            votes++;
        }
        else{
            if (i == candidate) votes++;
            else votes--;
        }
    }

    let count: number = 0;
    for(let i of nums){
        if (candidate == i) count++;
        
    }

    if (count > nums.length / 2) return candidate;
    else return -1;
};